content/blog/2020-05-05-updates-to-ragg-and-systemfonts/index.Rmarkdown

```diff
@@ -0,0 +1,169 @@
+
+```{r, message=FALSE}
+library(ragg)
+library(fs)
+
+# Creating some image data
+library(ambient)
+library(dplyr)
+noise <- long_grid(
+    x = seq(0, 1, length.out = 2000), 
+    y = seq(0, 1, length.out = 2000)
+  ) %>% mutate(
+    z = fracture(
+      noise = gen_worley, 
+      fractal = billow, 
+      octaves = 4, 
+      gain = ~ .x * 0.75,
+      frequency = ~ .x * 3,
+      x = x, 
+      y = y,
+      value = "distance",
+      seed = 12
+    )
+  )
+
+# plot using png
+pngfile <- path(knitr::fig_path(),  "png_image.png")
+agg_png(pngfile, 2000, 2000)
+plot(noise, z)
+invisible(dev.off())
+knitr::include_graphics(pngfile)
+```
+
+```{r}
+# plot using jpeg
+jpegfile <- path(knitr::fig_path(),  "jpeg_image.jpeg")
+agg_jpeg(jpegfile, 2000, 2000)
+plot(noise, z)
+invisible(dev.off())
+
+knitr::include_graphics(jpegfile)
+```
+
+As you can hopefully agree on, there is no noticeable quality difference between the two. What about file size?
+
+```{r}
+file_size(pngfile)
+file_size(jpegfile)
+```
+
+We observe more than a 50% file size reduction. Again, bear in mind that this is only when saving image-type plots with high-frequency variations. `agg_png()` should always be your go-to device for any standard type of data visualization.
+
+## systemfonts 0.2.0
+systemfonts is a package that spun out of the development of ragg. Its main claim-to-fame is its ability to find the correct font file for a specific font and style request. With this release we have fixed up a lot of the underlying implementation as well as added a bunch of new features that may interest you.
+
+### Why are text so hard?
+Before we delve into the features of systemfonts it may be good to understand why fonts doesn't just work. R's font support has always been somewhat lacking and a range of packages has tried to solve this, e.g. extrafont and sysfonts. At its core text and fonts are just very messy, with differences between operating systems and font file formats to name some of the challenges. For a humorous overview consult the [*Text Rendering Hates You*](https://gankra.github.io/blah/text-hates-you/) article and have a chuckle. There are simply so many issues related to how finding the font files, reading the content, figuring out the correct glyphs for a character string (which may be encoded as god knows what) and determining the correct position of each glyph, that most applications that does something with text uses a high-level solutions such as those provided by the operating system or e.g. [Pango](https://www.pango.org). With all these complications it is easy to forgive the graphic devices in R for not being complete, since they have to take care of all of this from scratch.
```

Comment:
```suggestion
Before we delve into the features of systemfonts it may be good to understand why fonts don't just work. At its core text and fonts are just very messy, with differences between operating systems and font file formats to name some of the challenges. (For a humorous overview consult the [*Text Rendering Hates You*](https://gankra.github.io/blah/text-hates-you/)). 

There are simply so many issues related to how finding the font files, reading the content, figuring out the correct glyphs for a character string, and determining the correct position of each glyph, that most applications use high-level solutions such as those provided by the operating system or e.g. [Pango](https://www.pango.org). All-in-all this makes it rather difficult to do fully correct text rendering on graphics device.
```